Upgraded jemallocator dependency

Razican opened the pull request:
This PR just upgrades jemalloc and criterion to the latest supported versions. It does not require any code change, and tests seem to pass. It updates the two missing dependencies after #1101 and #1073 (or #950) get merged.

The API does not change, so this can be released as a minor version.

Razican commented:
Issues are related to a new minimum rust version.

ErichDonGubler commented:
@Geal, ping!

Geal commented:
since this would rais the minimum rust version, the PR needs to go in nom 6 (although criterion could be updated separately)

Razican commented:
I rebased and modified the commits to upgrade only the "jemallocator" dependency, that is compatible with Rust 1.31.0, so this could be added to nom 0.5, @Geal.

Geal commented:
thanks
